### PR TITLE
Feat/angular material/boolean tests

### PR DESCRIPTION
**Pull request by eneufeld**
~This is based on #1081~ 
~The new tests are failing we should discuss what we want to fix~

Tests should be running now and I rebased on master

### CODE REVIEW

File: packages/angular-material/test/boolean-control.spec.ts [hunk 1]
```diff
@@ -22,15 +22,73 @@
...
+import { NOT_APPLICABLE } from '@jsonforms/core';
+import { BooleanControlRenderer, booleanControlTester } from '../src';
 
+const uischema = {
+    type: 'Control',
+    scope: '#/properties/foo'
+};
+
+describe('Material boolean field tester', () => {
+    it('should fail', () => {
```
Comment (edgarmueller) [inline packages/angular-material/test/boolean-control.spec.ts]: I don't think we need the 'it should fail' spec, as we already test these in `testers.test.ts`

File: packages/angular-material/test/boolean-control.spec.ts [hunk 2]
```diff
@@ -41,18 +99,16 @@ describe('Boolean control', () => {
...
-        scope: '#/properties/foo'
-    };
 
-    beforeEach(async (() => {
+    const getImports = () => [
+        MatCheckboxModule,
+        MatFormFieldModule
+    ];
+
+    beforeEach(async(() => {
```
Comment (edgarmueller) [inline packages/angular-material/test/boolean-control.spec.ts]: Do we need the `async`?

File: packages/angular-material/test/boolean-control.spec.ts [hunk 3]
```diff
@@ -96,23 +160,181 @@ describe('Boolean control', () => {
...
         });
         mockSubStore.complete();
+        fixture.detectChanges();
         component.ngOnInit();
-        MockNgRedux.reset();
-        const mockSubStore2 = MockNgRedux.getSelectorStub();
-        mockSubStore2.next({
+
+        expect(component.data).toBe(true);
+        expect(checkboxInstance.checked).toBe(true);
```
Comment (edgarmueller) [inline packages/angular-material/test/boolean-control.spec.ts]: I don't think we necessarily have to expect these (here and in the following tests) as we already covered that case

File: packages/angular-material/test/boolean-control.spec.ts [hunk 4]
```diff
@@ -96,23 +160,181 @@ describe('Boolean control', () => {
...
+        });
+        fixture.detectChanges();
+        component.ngOnInit();
+        expect(component.data).toBe(true);
+        expect(checkboxInstance.checked).toBe(true);
+
+        mockSubStore.next({
+            jsonforms: {
+                core: {
+                    data: { foo: true, bar: false },
```
Comment (edgarmueller) [inline packages/angular-material/test/boolean-control.spec.ts]: I'm not sure how the test name relates to the `data` here. Mind to clarify?

File: packages/angular-material/test/boolean-control.spec.ts [hunk 5]
```diff
@@ -96,23 +160,181 @@ describe('Boolean control', () => {
...
+        fixture.detectChanges();
+        expect(component.data).toBe(true);
+        expect(checkboxInstance.checked).toBe(true);
+    });
+    it('can be disabled', () => {
+        const mockSubStore = MockNgRedux.getSelectorStub();
+        component.uischema = uischema;
+        component.disabled = true;
+
+        mockSubStore.next({
```
Comment (edgarmueller) [inline packages/angular-material/test/boolean-control.spec.ts]: We don't think we need the store at all in this test case.

File: packages/angular-material/test/boolean-control.spec.ts [hunk 6]
```diff
@@ -96,23 +160,181 @@ describe('Boolean control', () => {
...
+        component.ngOnInit();
+        expect(checkboxInstance.disabled).toBe(true);
+
+    });
+    it('id should be present in output', () => {
+        const mockSubStore = MockNgRedux.getSelectorStub();
+        component.uischema = uischema;
+        component.id = 'myId';
+
+        mockSubStore.next({
```
Comment (edgarmueller) [inline packages/angular-material/test/boolean-control.spec.ts]: Here as well.

File: packages/angular-material/test/boolean-control.spec.ts (hunk 1 above)
Comment (eneufeld) [reply packages/angular-material/test/boolean-control.spec.ts]: it makes sure that the wiring was done correctly, but I'm all for removing those tests, the hard question is what do we need to retest when testing wiring and what not

File: packages/angular-material/test/boolean-control.spec.ts (hunk 3 above)
Comment (eneufeld) [reply packages/angular-material/test/boolean-control.spec.ts]: yes, we could leave them out

File: packages/angular-material/test/boolean-control.spec.ts (hunk 4 above)
Comment (eneufeld) [reply packages/angular-material/test/boolean-control.spec.ts]: the idea is to test, that only changes on the relevant field are reflected, probably doesn't make much sense here, and should only be tested for the mapToControlProps

File: packages/angular-material/test/boolean-control.spec.ts (hunk 5 above)
Comment (eneufeld) [reply packages/angular-material/test/boolean-control.spec.ts]: probably not, I will recheck

**Comment by eneufeld** [comment]
@edgarmueller should I do the fixes here or would you be fine with a clean up commit?

**Comment by edgarmueller** [comment]
@eneufeld Yes, I'm fine with that.

**Pull request merged**